docker_ci_deploy/tests/test_main.py

```diff
@@ -4,67 +4,156 @@
 import sys
 from subprocess import CalledProcessError
 
+import pytest
 from testtools import ExpectedException
 from testtools.assertions import assert_that
 from testtools.matchers import (
-    AfterPreprocessing as After, Equals, MatchesRegex, MatchesStructure, Not)
+    AfterPreprocessing as After, Equals, Is, MatchesRegex, MatchesStructure,
+    Not)
 
 from docker_ci_deploy.__main__ import (
-    cmd, DockerCiDeployRunner, main, strip_image_tag)
+    cmd, DockerCiDeployRunner, join_image_tag, main, replace_image_registry,
+    replace_tag_version, split_image_tag)
 
-""" strip_image_tag() """
+""" split_image_tag() """
 
 
-def test_strip_image_tag():
+def test_split_image_tag():
     """
-    Given an image tag with registry, name and tag components, strip_image_tag
-    should strip only the tag component.
+    Given an image tag with registry, name and tag components, split_image_tag
+    should return the registry and name as the image and the tag part as the
+    tag.
     """
-    image = 'registry.example.com:5000/user/name:tag'
-    stripped_tag = strip_image_tag(image)
+    image_tag = 'registry.example.com:5000/user/name:tag'
+    image, tag = split_image_tag(image_tag)
 
-    assert_that(stripped_tag, Equals('registry.example.com:5000/user/name'))
+    assert_that(image, Equals('registry.example.com:5000/user/name'))
+    assert_that(tag, Equals('tag'))
```

Comment:
/shrug I could if that's better?
It would be something like this:
```python
image_tag = split_image_tag('registry.example.com:5000/user/name:tag')
assert_that(image_tag, MatchesListwise([
  Equals('registry.example.com:5000/user/name')
  Equals('tag')
]))
```